#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Correct pthread_create manpage to match texinfo documentation

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- linuxthreads/man/pthread_create.man.orig	Tue Sep 21 16:12:25 1999
+++ linuxthreads/man/pthread_create.man	Tue Sep 21 16:15:04 1999
@@ -31,10 +31,8 @@
 .SH ERRORS
 .TP
 !EAGAIN!
-not enough system resources to create a process for the new thread.
-.TP
-!EAGAIN!
-more than !PTHREAD_THREADS_MAX! threads are already active.
+not enough system resources to create a process for the new thread,
+or more than !PTHREAD_THREADS_MAX! threads are already active.
 
 .SH AUTHOR
 Xavier Leroy <Xavier.Leroy@inria.fr>
